### DISCUSSION TITLE
Possible to adjust speed?

### DISCUSSION THREAD

**Discussion by jesperlandberg** [idiotWu/smooth-scrollbar#350] [Q&A] [answered]
Hi,
Thanks for a good plugin.

Is it possible to adjust the speed of the scrolling also? not only damping.

**Comment by dolphin-wood** [comment, marked as answer]
For smooth-scrollbar@8.x, you can use the [plugin system](https://github.com/idiotWu/smooth-scrollbar/blob/develop/docs/plugin.md) to do this:

```js
import Scrollbar, { ScrollbarPlugin } from 'smooth-scrollbar';

class ScaleSpeedPlugin extends ScrollbarPlugin {
  static pluginName = 'scaleSpeed';

  static defaultOptions = {
    speed: 1,
  };

  transformDelta(delta) {
    const { speed } = this.options;

    return {
      x: delta.x * speed,
      y: delta.y * speed,
    };
  }
}

Scrollbar.use(ScaleSpeedPlugin);

Scrollbar.init(elem, {
  plugins: {
    scaleSpeed: {
      speed: 10,
    },
  },
});
```

**Reply by freestyle099** [reply]
You forgot that your class needs to extend `ScrollbarPlugin` only then it works

**Reply by freestyle099** [reply]
And thank you for an answer 😄

**Reply by freestyle099** [reply]
Ahh and in Scrollbar.use() should be `ScaleSpeedPlugin` not `ScalePlugin`
